src/common/api/hive.ts

```diff
@@ -586,14 +586,39 @@ export const getSavingsWithdrawFrom = (account: string): Promise<SavingsWithdraw
 export interface BlogEntry {
   blog: string;
   entry_id: number;
+  post_id?: number;
+  num?: number;
   author: string;
   permlink: string;
   reblogged_on: string;
+  created?: string;
 }
 
 export const getBlogEntries = (username: string, limit: number = dataLimit): Promise<BlogEntry[]> =>
   client.call("condenser_api", "get_blog_entries", [username, 0, limit]);
 
+export const getAccountVotesTrail = (
+  username: string,
+  start = -1,
+  limit = 20
+): Promise<BlogEntry[]> => {
+  return getAccountHistory(
+    username,
+    utils.makeBitMaskFilter([utils.operationOrders.vote]),
+    start,
+    limit
+  ).then((data) => {
+    let result = data
+      .map((obj: any) => {
+        return { ...obj[1].op[1], num: obj[0] };
+      })
+      .filter((obj: any) => obj.voter === username);
+    return Promise.all(
+      result.map((obj: any) => getPostNew(obj.author, obj.permlink, username, obj.num))
```

Comment:
Please change the variable name obj to some better understandable name. From 612 - 617 obj is used in 3 different functions. May be confusing in the future. Please use different names in all functions if possible